open-api/rest-catalog-open-api.yaml

```diff
@@ -541,6 +541,216 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Returns either a list of `PlanTask`, a list of `FileScanTask`. If planning is not complete returns a `plan-id`.
+      description:
+        Prepares either a list of `PlanTask` that can be used to distribute table scan planning, or a list of `FileScanTask`, based on a set of table scan criteria
+        such as selected columns, filters, snapshot range, case sensitivity, etc.
+        In the event that the plan tasks or file scan tasks are not ready to be served, the service will return a `plan-id`,
```

Comment:
The `plan-id` can be returned even if plan tasks are returned.  I don't really think it makes sense if to return an id if only file scan tasks are present since all the planning is complete at that point.

Reply:
I think to simply this, we should say the server can return any combination of plan-id, planTasks, fileTasks.  If a plan-id is returned, get plan status should be called until the result is in a terminal state (finished/cancelled/failed).

Reply:
Thanks @danielcweeks for this feedback however I thought @rdblue mentioned that returning a `plan-id` is the case that your plan tasks, and file scan tasks have not finished processing, thus you are given this id to continuously poll with the `GetTasksStatus` api until they are finished. To me it seems somewhat complicated to now have id be returned with the tasks as well. 

Curious what others think @amogh-jahagirdar @jackye1995

Reply:
Yeah, I think what the issue was is that if you return Plan Tasks, there may still be planning going on that needs to be cancelled, so you can return a `plan-id` along with other results and cancellation ideally would call cancel on the `plan-id`.  Maybe that's too complicated and we should stick with either `plan-id` or `plan/file tasks`. (thoughts @rdblue?)

Reply:
To me I think the idea of returning a partial amount of plan tasks with an id makes this protocol a little more complicated for two reasons.

1. To me I thought the goal was to make sure that planning is fully done before returning any planTasks such that you get good gurantees that work can be distributed evenly (as you can see how many file scan tasks are generated on service side and allocate appropriate amount of file scan tasks per plan task)

2. I feel that just returning plan tasks file scan tasks without an id, is the best signal to indicate that planning is done. Returning the `plan-id` by itself signals that planning is not finished and that a person can either poll using this id for their tasks with `GetTasksStatus` , or go and cancel the plan entirely with this id in `CancelPlan`

Reply:
I clarified this behavior in two ways:
1. I made it so that status is always required so it is clear when cancellation is necessary. With a "success" status, the service is allowed to return plan-tasks and file-scan-tasks, but not at other times. Plan-id can always be sent.
2. The cancellation endpoint clarifies when it should be called with a plan-id. If the plan-id has returned "success" then it should be called if the client will not fetch scan tasks for all of the plan tasks.